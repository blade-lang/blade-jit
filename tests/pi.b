# For now, this is just good enough...
# the error margin is very minimal!
def pi(K) {
  var A = 545140134, B = 13591409, D = 640320
  var id3 = 1.0 / (D ** 3)

  var sum = 0.0, b = id3 ** 0.5, p = 1.0, a = B
  sum = p * a * b

  iter var k = 1; k < K; k++ {
    # A * k + B
    a += A

    # update denominator
    b *= id3
    p *= (6 * k) * (6 * k - 1) * (6 * k - 2) * (6 * k - 4) * (6 * k - 5)
    p /= (3 * k) * (3 * k - 1) * (3 * k - 2) * (k ** 3)
    p -= p

    sum += p * a * b
  }

  return 1.0 / (12 * sum)
}

echo pi(5)
