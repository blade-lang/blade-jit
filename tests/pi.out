3.141592653589734